message_formatting: Improve the message formatting tips table.

sbansal1999 opened the pull request:
The CSS class `spacer` was required or else there was very little spacing between the input elements in the `/poll` widget. 

<table>
  <thead>
    <tr>
      <th colspan="2">Use of <q>spacer</q> class</th>
    </tr>
    <tr>
      <th>Before</th>
      <th>Header</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><img src="https://github.com/zulip/zulip/assets/35286603/9db7d74e-86b2-49f9-bfd5-46ee157ff1aa" alt="image"></td>
      <td><img src="https://github.com/zulip/zulip/assets/35286603/d840cdc0-fdab-4f4f-af2f-16febaeb01c2" alt="image"></td>
    </tr>
  </tbody>
</table>


Fixes: #25780


<details>
<summary>Screenshots:</summary>

| Before | After |
|--------|--------|
| ![Screenshot from 2023-05-27 16-01-50](https://github.com/sbansal1999/zulip/assets/35286603/efd98a53-4fec-47b9-8e81-c1b1aa5ff855) | ![Screenshot from 2023-05-27 16-01-05](https://github.com/sbansal1999/zulip/assets/35286603/f60be2f6-1909-4d7c-bfec-2e388996e715) |
| ![Screenshot from 2023-05-27 16-01-52](https://github.com/sbansal1999/zulip/assets/35286603/13370f6a-985b-4da1-a309-f1e091a0f837) | ![Screenshot from 2023-05-27 16-10-56](https://github.com/sbansal1999/zulip/assets/35286603/0ce6f9b6-aada-4c9e-b022-de08f26c64de) | 


</details>

<hr>

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

sbansal1999 commented:
@zulipbot add "buddy review"

akshatdalton commented:
LGTM

sbansal1999 commented:
@zulipbot remove "buddy review"
@zulipbot add "maintainer review"

amanagr commented on web/styles/widgets.css:
```diff
@@ -27,6 +27,11 @@
         margin-right: 4px;
     }
 
+    /* for example in help `?` widget */
+    .spacer {
+        margin: 2px 0 !important;
+    }
+
```
Can we use a class name specific to that widget if we only want to fix that.

sbansal1999 replied:
Yeah, that makes sense as `spacer` is too generic.

sbansal1999 replied:
Changed it to `todo-task-spacer`.

amanagr commented:
@sbansal1999 thanks for working on this! Left a comment.

sbansal1999 commented:
Thanks for the review @amanagr. :heart_decoration: 

I have made the requested changes, let me know if there is anything more required in this one.

amanagr commented:
LGTM!

timabbott commented on web/src/info_overlay.js:
```diff
@@ -150,6 +166,23 @@ Coffee`,
+         </div>
+      </div>
+   </div>
+</div>
```
So I think we want the `/todo` example to be one where we pass more details when making it, like we do the poll, so that you can see what it will look like.

sbansal1999 replied:
Yeah, that makes sense. I have updated the code to now include two todos; one of them has been marked as checked and hence is crossed over.

![image](https://github.com/zulip/zulip/assets/35286603/5bb583ab-5020-4153-ba98-d8511c7642da)

The only minor issue is that ticking/unticking the checkboxes has no effect on the task.

timabbott replied:
Yeah that's probably OK.

sbansal1999 replied:
Makes sense.

timabbott commented on web/src/info_overlay.js:
```diff
@@ -120,6 +130,12 @@ def zulip():
+        output_html:
+            '<p><time datetime="2023-05-28T08:00:00Z"><i class="fa fa-clock-o"></i>Sun, May 28, 2023, 1:30 PM</time></p>',
+        effect_html: "(displayed to everyone in their own time zone)",
+    },
```
I think after seeing this, let's skip the `effect_html`; you get the same information by hovering over it with the mouse, and it feels pretty ugly with a wrapped line there.

sbansal1999 commented:
@timabbott Thanks for reviewing my work. I have updated the PR and made the requested changes. Also dropped one minor issue in the comment.

Changes made:

- Dropped the `effect_html` for the date time message.
- Added a couple of sample todos to the todo widget.

sbansal1999 commented:
**Updates:**

- Removed the `New task` and `Description` fields and the `Add task` button from the table.
- Also had to delete the `todo-task-spacer` class as it was no longer required.

Updated Screenshot:

![image](https://github.com/zulip/zulip/assets/35286603/28c14acc-4213-4f2c-913b-97548695a833)

timabbott commented:
OK I made a few tweaks to the text for the /todo one, including hiding the heading that looks ugly here. 

With those changes, this looks great, marked to merge, thanks @sbansal1999!

sbansal1999 commented:
@timabbott It looks like you didn't push the changes you mentioned in the previous comment.

timabbott commented:
Indeed, just noticed getting back to my computer that the push failed, going to just add a follow-up.

timabbott commented:
Pushed 654fb188327cf9d22fe27118c6a531146be5e247.
